block/manager.go

```diff
@@ -212,13 +217,25 @@ func (m *Manager) healthStatusEventCallback(event pubsub.Message) {
 }
 
 func (m *Manager) applyBlockCallback(event pubsub.Message) {
-	m.logger.Debug("Received new block event", "eventData", event.Data())
+	m.logger.Debug("Received new block event", "eventData", event.Data(), "cachedBlocks", len(m.prevBlock))
 	eventData := event.Data().(p2p.GossipedBlock)
 	block := eventData.Block
 	commit := eventData.Commit
-	err := m.applyBlock(context.Background(), &block, &commit, blockMetaData{source: gossipedBlock})
-	if err != nil {
-		m.logger.Debug("Failed to apply block", "err", err)
+
+	if block.Header.Height != m.store.Height()+1 {
+		// We crashed after the commit and before updating the store height.
+		m.prevBlock[block.Header.Height] = &block
+		m.prevCommit[block.Header.Height] = &commit
+		m.logger.Debug("Caching block", "block height", block.Header.Height, "store height", m.store.Height())
+		err := m.checkPrevCachedBlocks(context.Background())
+		if err != nil {
+			m.logger.Debug("Error applying previous cached blocks", "err", err)
```

Comment:
Im not sure if `Error` is correct here as it’s not necessarily due to error - it’s mostly based on attempting. So maybe “failed” . especially when you log it with a "debug"

Reply:
log message changed